SIP-0064: Remove Bridge Functions from Mynt (#64)
DistributedCollective/SIPS: SIP-0064.md

---
SIP: '0064'
Title: Remove Bridge Functions from Mynt
Author: Tyrone Johnson (@tjcloa)
Status: Draft
Track: Contract
Created: 2023-06-07
---

# SIP-0064: Remove Bridge Functions from Mynt

## Description  

> There are two cases of the Mynt contracts upgrades included in this SIP:  

Reviewer: ```suggestion
There are two Mynt contract updates included in this SIP:  
```

> - Removal of legacy bridges related code 

Reviewer: ```suggestion
- Removal of legacy code related to the Svoryn Bridge integration  
```

> - Technical upgrade of the contracts with the bytecode changed due to NatSpec comments modified   

Reviewer: ```suggestion
- Technical update of the contracts with the bytecode changed due to modified NatSpec comments  
```

  
### 1) Removal of legacy bridges related code

>     Mynt was initially cloned from Babelfish protocol as the basis for Sovryn DLLR BTC-based stablecoin.     
>     Babelfish uses bridges as trusted sources for convertions. Howerver it was decided not to use the bridges in Mynt.  
>     To stay on schedule bridges related functionality was muted and unused by the time of the first release and which created a so-called 'technical debt' which we are paying back now.  
> 
>     Unused code in smart contracts not only causes confusion but also is a potential source of bugs and vulnerabilities in the future.  
>     Following smart contracts security best practices we have removed all the code related to bridges.  
> 

Reviewer: ```suggestion
Mynt was initially cloned from Babelfish protocol as the basis for Sovryn DLLR BTC-based stablecoin.     

Babelfish uses bridges as trusted sources for conversions. However it was decided not to use the bridges in Mynt.  

To keep Mynt development on schedule, the bridge-related functionality was muted and unused, which created technical debt that we are paying back now with this SIP.  

Unused code in smart contracts not only causes confusion for reviewers and integrators but is also a potential source of bugs and vulnerabilities in the future.  

Following smart contract security best practices we have removed all the code related to bridge integration.  

```

### 2) Technical upgrade

>     Changes in OpenZeppelin (OZ) contract `Initializable` inherited by `FeesVault` and `MocIntegration` contracts: OZ package version upgrade to 4.8.3 among other fixes changed NatSpec functions comments which are added by a solidity compiler as metadata to the bytecode of the FeesVault and MocIntegration contracts: 

Reviewer: ```suggestion
Changes in OpenZeppelin (OZ) contract `Initializable` inherited by `FeesVault` and `MocIntegration` contracts: OZ package version upgrade to 4.8.3 among other fixes changed the NatSpec function comments, which are added by a solidity compiler as metadata to the bytecode of the FeesVault and MocIntegration contracts: 
```

&nbsp;&nbsp;&nbsp;<img src="images/FeesVault diff example.png" alt="FeesVault Contract NatSpec Comment Change" width="80%"/>

> __If approved, this proposal will upgrade Mynt contracts affected by the code cleanup: BasketManagerV3, MassetManager, FeesManager, FesVault and MocIntegration.__ 

Reviewer: ```suggestion
__If approved, this proposal will update the following Mynt contracts affected by the code cleanup: BasketManagerV3, MassetManager, FeesManager, FeesVault, and MocIntegration.__ 
```
